services/core/events/blockchain.js

```diff
@@ -49,14 +47,15 @@ module.exports = [
 						util.inspect(data),
 					].join('\n'));
 				}
-			});
+			};
+			Signals.get('newBlock').add(newBlockListener);
```

Comment:
Is there any reason to wrap the given logic in function before calling? (Ideally, both logic seems to be same)

Reply:
Separate instances of similar anonymous or named functions within Javascript when strictly compared (`===`) evaluate to `false`. Therefore, for `Signals.has()` to evaluate to `true`, we need named listener functions. `has()` internally performs a strict comparison to evaluate the registration of a listener function.